jetbrains-core/src/software/aws/toolkits/jetbrains/ui/AsyncComboBox.kt

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.ui.ComboBox
+import com.intellij.ui.AnimatedIcon
+import com.intellij.ui.SimpleListCellRenderer
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.annotations.TestOnly
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.spinUntil
+import software.aws.toolkits.jetbrains.utils.ui.selected
+import java.awt.Component
+import java.time.Duration
+import java.util.Timer
+import java.util.TimerTask
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.swing.DefaultComboBoxModel
+import javax.swing.JList
+import javax.swing.event.ListDataEvent
+import kotlin.concurrent.timerTask
+
+class AsyncComboBox<T>(
+    private val comboBoxModel: DefaultComboBoxModel<T> = DefaultComboBoxModel(),
```

Comment:
Does this need to be Default, or can it be any `javax.swing.MutableComboBoxModel`

Reply:
was default for convenience, will change